examples/zephyr/hello/pytest/test_sample.py

```diff
@@ -11,15 +10,7 @@
 
 pytestmark = pytest.mark.anyio
 
-async def test_hello(shell, api_key, device_name, credentials_file):
-
-    # Connect to Golioth and get device object
-
-    client = Client(api_url = "https://api.golioth.dev",
-                    api_key = api_key)
-    project = (await client.get_projects())[0]
-    device = await project.device_by_name(device_name)
-
+async def test_hello(shell, device, credentials_file):
```

Comment:
I don't understand where `device` is coming from.

I expected to see a change in the workflow that passes this in but that's not the case.

Reply:
`device` is provided by the new pytest plugin that was added to `python-golioth-tools`. The change in the workflow is subtle - it's the update to the revision of the python package that we install as part of the workflow. This is one of those things that's both slick and frustrating about pytest plugins. Slick because all a testcase needs to do is add an argument to its function and presto, pytest provides that argument. Frustrating because all that magic makes it non-obvious what's going on under the hood.